qiskit/circuit/oracle/types.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```

Comment:
let's move these types one level up to be under `circuit`. They are not really "oracle types", they're just some data types representing numbers.

Actually I should be able to import `Int1` the same way I import Qubit/Clbit now, i.e. `from qiskit.circuit import Int1`

Reply:
These types are not like functional, unlike qubit/clbit. They are just for the annotation. Do you still want them at the same level?

Reply:
done!